src/Streamly/Internal/LZ4.hs

```diff
@@ -422,18 +422,22 @@ resizeD Config{..} (Stream.Stream step0 state0) =
             Stream.Skip st1 -> return $ Stream.Skip $ RAccumlate st1 buf
             Stream.Stop -> error "resizeD: Incomplete block"
     step gst (RFooter st buf) = do
-        r <- step0 gst st
-        case r of
-            Stream.Yield arr st1 -> do
-                arr1 <- Array.spliceTwo buf arr
-                return $ Stream.Skip $ RFooter st1 arr1
-            Stream.Skip st1 -> return $ Stream.Skip $ RFooter st1 buf
-            Stream.Stop -> do
-                -- Validate footer here
-                res <- liftIO $ validateFooter buf
-                if res
-                then return Stream.Stop
-                else error "resizeD: Invalid footer"
+        -- Warn if len > footerSize
+        let len = Array.byteLength buf
+        if len < footerSize
+        then do
+            r <- step0 gst st
+            case r of
+                Stream.Yield arr st1 -> do
+                    arr1 <- Array.spliceTwo buf arr
+                    return $ Stream.Skip $ RFooter st1 arr1
+                Stream.Skip st1 -> return $ Stream.Skip $ RFooter st1 buf
+                Stream.Stop -> error "resizeD: Incomplete footer"
+        else do
+            res <- liftIO $ validateFooter buf
+            if res
+            then return Stream.Stop
+            else error "resizeD: Invalid footer"
```

Comment:
validateFooter should anyway print a relevant error.

Reply:
This would require some enhancements or we can assume that an invalid footer is only because of checksum mismatch.
Considering that we will rewrite this anyway, I'm not going to pursue this.